api/responses/ok.js

```diff
@@ -25,9 +26,9 @@ module.exports = function sendOK(data, options) {
 
   // If appropriate, serve data as JSON(P)
   var JSONresponse = {
-    links: {
+    links: _.extend({
       self: req.baseUrl + req.url
-    },
+    }, links),
```

Comment:
`req.baseUrl` should probably be prepended to all link urls.
to be even more efficient one could put the `self` link through some sort of magic between controller + response so that it happens everywhere, not just on `ok` - that's a very much 'nice to have' if you have the energy and inclination to do it - it's not likely to hurt us yet